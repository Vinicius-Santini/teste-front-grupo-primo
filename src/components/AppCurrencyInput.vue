<script setup>
import { makeCurrencyMask } from '../helpers';

const props = defineProps(['modelValue'])

const emit = defineEmits(['update:modelValue'])

function handleInputedValue(event) {
    const maskedValue = makeCurrencyMask(event.target.value)
    event.target.value = maskedValue
    const floatString = maskedValue.replace(/\./g, '').replace(/,/g, '.').replace("R$Â ", "")
    const investmentValue = parseFloat(floatString)
    emit('update:modelValue', investmentValue)
}
</script>

<template>
    <input type="text" placeholder="R$ 0,00" @input="handleInputedValue($event, inputName)">
</template>