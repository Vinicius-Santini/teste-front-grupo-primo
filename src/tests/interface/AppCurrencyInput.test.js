import { test, expect } from "vitest";
import { render } from "@testing-library/vue";
import AppCurrencyInput from "../../components/AppCurrencyInput.vue";

test("AppCurrencyInput should render correctly", () => {
    const appCurrencyInput = render(AppCurrencyInput)

    expect(AppCurrencyInput).toMatchSnapshot();
})