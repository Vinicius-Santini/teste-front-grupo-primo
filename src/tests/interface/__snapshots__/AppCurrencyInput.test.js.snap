// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppCurrencyInput should render correctly 1`] = `
{
  "__file": "C:/Users/vinic/Desktop/Projeto/teste-front-grupo-primo/src/components/AppCurrencyInput.vue",
  "__name": "AppCurrencyInput",
  "emits": [
    "update:modelValue",
  ],
  "props": [
    "modelValue",
  ],
  "render": [Function],
  "setup": [Function],
}
`;
